scripts/generate_maps.py

```diff
@@ -0,0 +1,177 @@
+
+        # Populate the dictionary
+        for layer in layers:
+            try:
+                gdf = gpd.read_file(input_file_path, layer=layer)  # Read the full layer
+                if 'institution' in gdf.columns:
+                    institutions = gdf['institution'].unique()
+                    for institution in institutions:
+                        if institution not in institution_layers:
+                            institution_layers[institution] = []
+                        institution_layers[institution].append(layer)
+            except Exception as e:
+                print(f"Error reading layer {layer}: {e}")
+
+        print("Institution layers mapping:")
+        print(institution_layers)
+
+        # Create an overview GeoDataFrame
+        overview_gdf = gpd.GeoDataFrame()
+
+        # Iterate through each institution and create maps
+        for institution, layers in institution_layers.items():
+            institution_data = []
+
+            print(f"\nProcessing institution: {institution}")
+
+            for layer in layers:
+                print(f"Checking layer: {layer}")
+                try:
+                    gdf = gpd.read_file(input_file_path, layer=layer)  # Read the full layer
+
+                    if 'institution' in gdf.columns and institution in gdf['institution'].unique():
+                        print(f"Layer '{layer}' contains '{institution}'.")
+                        institution_data.append(gdf[gdf['institution'] == institution])
+                    else:
+                        print(f"Layer '{layer}' does not contain '{institution}' or does not have 'institution' column.")
+                except Exception as e:
+                    print(f"Error processing layer {layer}: {e}")
+
+            # Combine all dataframes for the institution
+            if institution_data:
+                institution_gdf = gpd.GeoDataFrame(pd.concat(institution_data, ignore_index=True))
+            else:
+                institution_gdf = gpd.GeoDataFrame()
+
+            # Ensure the CRS matches between the basemap and the GeoDataFrame
+            if not institution_gdf.empty:
+                institution_gdf = institution_gdf.to_crs(mp.crs)
+
+            # Add to the overview GeoDataFrame
+            if not institution_gdf.empty:
+                overview_gdf = pd.concat([overview_gdf, institution_gdf])
+
+            # Plot and save the aggregated data for the institution
+            plot_data_on_basemap(mp, institution_gdf, institution, f'Antarctica_coverage_{institution}.png')
+
+        # Ensure the CRS matches between the basemap and the overview GeoDataFrame
+        if not overview_gdf.empty:
+            overview_gdf = gpd.GeoDataFrame(overview_gdf).to_crs(mp.crs)
+
+        # Plot and save the overview map
+        plot_data_on_basemap(mp, overview_gdf, "Overview", 'Antarctica_coverage_overview.png')
+
+        print("All maps have been created and saved to the data3 folder.")
+
+else:
```

Comment:
I'd switch the order of the if/else statement -- better to handle error and exit early at the start, rather than have to read all the way to the bottom to see this one-liner.